src/argilla/server/search_engine/elasticsearch.py

```diff
@@ -98,7 +98,7 @@ async def _request_similarity_search(
             # TODO: Add metadata !!!!
             knn_query["filter"] = self._build_response_status_filter(user_response_status_filter)
 
-        return await self.client.search(index=index, knn=knn_query, _source=False, track_total_hits=True)
+        return await self.client.search(index=index, knn=knn_query, _source=False, track_total_hits=True, size=k)
```

Comment:
It is possible to rename `k` to `size`? If `k` has some clear meaning (like in k-clustering or something) in the context of this function then forget about this suggestion.

```suggestion
        return await self.client.search(index=index, knn=knn_query, _source=False, track_total_hits=True, size=size)
```

Reply:
Related capabilities for OpenSearch and Elasticsearch refer explicitly to the `k` parameter. I would keep it aligned since `size` may be confusing in this context.